src/cargo/core/compiler/mod.rs

```diff
@@ -1313,7 +1313,11 @@ fn sysroot_remap(build_runner: &BuildRunner<'_, '_>, unit: &Unit) -> OsString {
     let sysroot = &build_runner.bcx.target_data.info(unit.kind).sysroot;
     let mut remap = OsString::from("--remap-path-prefix=");
     remap.push(sysroot);
-    remap.push("/lib/rustlib/src/rust"); // See also `detect_sysroot_src_path()`.
```

Comment:
[`detect_sysroot_src_path`](https://github.com/rust-lang/cargo/blob/master/src/cargo/core/compiler/standard_lib.rs#L200-L206) achieves the same thing with `.join` on `sysroot` so I think this should be:
```rust
    remap.push(
        // See also `detect_sysroot_src_path()`.
        sysroot
            .join("lib")
            .join("rustlib")
            .join("src")
            .join("rust")
            .as_os_str(),
    );
```